test/compilable/scopeinfer.d

```diff
@@ -0,0 +1,12 @@
+// PERMUTE_ARGS: -dip1000
+
+// Mangling should be the same with or without inference of `return scope`
+
+@safe:
+
+auto foo(void* p) { return 0; }
+static assert(typeof(foo).mangleof == "FNaNbNiNfPvZi");
+
+auto bar(void* p) { return p; }
+static assert(typeof(bar).mangleof == "FNaNbNiNfPvZQd");
```

Comment:
Please also add tests that shows explicit annotations make it to the mangling.

Reply:
see testscope2.d

Reply:
testscope2.d only runs with -dip25, but some of the code only affects -dip1000 AFAICT.

Reply:
BTW: you can also use core.demangle with CTFE for more readable asserts.